packages/core/assistant/src/session/session.ts

```diff
@@ -188,52 +202,54 @@ export class AiSession {
     // return parser.getResult(result);
   }
 
-  private _formatUserPrompt = (prompt: string, history: DataType.Message[]) =>
-    Effect.gen(function* () {
-      const prelude: ContentBlock.Any[] = [];
-
-      // TODO(dmaretskyi): Evaluate other approaches as `serviceOption` isn't represented in the type system.
-      const artifactDiffResolver = yield* Effect.serviceOption(ArtifactDiffResolver);
-      if (Option.isSome(artifactDiffResolver)) {
-        const versions = gatherObjectVersions(history);
-
-        const artifactDiff = yield* Effect.tryPromise({
-          try: () =>
-            artifactDiffResolver.value.resolve(
-              [...versions.entries()].map(([id, version]) => ({ id, lastVersion: version })),
-            ),
-          catch: AiAssistantError.wrap('Artifact diff resolution error'),
-        });
+  // TODO(burdon): Implement abort.
+  abort(): void {
+    // this._stream?.abort();
+    // Effect.interrupt?
+  }
+}
 
-        log.info('vision', {
-          artifactDiff,
-          versions,
-        });
+const formatUserPrompt = (prompt: string, history: DataType.Message[]) =>
```

Comment:
Considering moving this to aipreprocessor, but you can leave this as a todo for me